clustermesh-apiserver/clustermesh/root.go

```diff
@@ -97,13 +99,53 @@ func registerHooks(lc cell.Lifecycle, params parameters) error {
 				return err
 			}
 
-			startServer(ctx, params.ClusterInfo, params.EnableExternalWorkloads, params.Clientset, backend, params.Resources, params.StoreFactory, params.SyncState)
+			startServer(ctx, params.CrdSyncPromise, params.ClusterInfo, params.EnableExternalWorkloads, params.Clientset, backend, params.Resources, params.StoreFactory, params.SyncState)
 			return nil
 		},
 	})
 	return nil
 }
 
+type syncCRDsPromiseParams struct {
+	cell.In
+
+	Lifecycle cell.Lifecycle
+	Clientset k8sClient.Clientset
+	Resources *synced.Resources
+	APIGroups *synced.APIGroups
+}
+
+func newSyncCRDsPromise(params syncCRDsPromiseParams) promise.Promise[resource.CRDSync] {
```

Comment:
Why is this in this package? I find it confusing that we have something here that is used pretty globally in the daemon and it's providing a type that it itself doesn't own. I would put this into `daemon/k8s` instead.